### PR TITLE
variant: G0: add Nucleo-G070RB

### LINKED ISSUE #1147: [New Variant] add support for Nucleo64 STM32G070
I have no idea why this board is not supported while in the main site clearly refrenced to being supported by arduino

 - Name: Nucleo64 STM32G070
 - Link to all relevant information:
   * Documentations : https://www.st.com/resource/en/user_manual/dm00452640-stm32-nucleo64-boards-with-stm32g07xrb-mcus-stmicroelectronics.pdf
   * Schematics :https://www.manualslib.com/manual/1529028/St-Stm32-Nucleo-64.html?page=11
   * Where this hardware can be purchased :https://www.st.com/en/evaluation-tools/nucleo-g070rb.html#:~:text=The%20STM32%20Nucleo%2D64%20boards,provided%20by%20the%20STM32%20microcontroller.

### LINKED ISSUE #1651: Nucleo-G070RB
**Board description (please complete the following information):**
 - Name: [e.g. DISCOVERY F769NI]
 - Link to all relevant information:
   * Documentations
   * Schematics
   * **[Where this hardware can be purchased]::-(https://www.st.com/en/evaluation-tools/nucleo-g070rb.html)**
  * ...

**Note that user can add a STM32 based board following this [wiki](https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-(board))**

Any contribution is welcome, so do not hesitate to submit a PR.

All new variant request will be moved to [New Variant] request list #722 and the related issue will be closed.

### PR DESCRIPTION
**Pull request by TDhaouST**
[Nucleo_g070rb
](https://www.st.com/en/evaluation-tools/nucleo-g070rb.html)
### **Tested:**
 

- GPIO
- U(S)ART
- ADC
- DAC
- I2C
- SPI
- Timer
- RTC
- Low Power
- EEPROM

Fixes #1147 and fixes #1651

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart:  | STM32F401RE | [Nucleo F401RE](http://www.st.com/en/evaluation-tools/nucleo-f401re.html) | *0.2.1* |  |
 | :green_heart:  | STM32F411RE | [Nucleo F411RE](http://www.st.com/en/evaluation-tools/nucleo-f411re.html) | *0.2.1* |  |
 | :green_heart:  | STM32F446RE | [Nucleo F446RE](http://www.st.com/en/evaluation-tools/nucleo-f446re.html) | *1.1.1* |  |
+|:green_heart:  | STM32G070RB | [Nucleo G070RB](https://www.st.com/en/evaluation-tools/nucleo-g070rb.html) | *1.6.0* |  |
```
Comment (fpistm) [inline README.md:114]:
```suggestion
|  :yellow_heart:  | STM32G070RB | [Nucleo G070RB](https://www.st.com/en/evaluation-tools/nucleo-g070rb.html) | **2.3.0** |  |
```

File: README.md [hunk 2]
```diff
@@ -275,6 +276,8 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
+| :yellow_heart: | STM32F722RC<br>STM32F722RE | Generic Board | **2.3.0** |  |
+| :yellow_heart: | STM32F730R8<br>STM32F732RE | Generic Board | **2.3.0** |  |
```
Comment (fpistm) [inline README.md:280]: This should not be there

```suggestion
```

File: variants/STM32G0xx/G070RBT/generic_clock.c [hunk 3]
```diff
@@ -20,12 +20,13 @@
   */
 WEAK void SystemClock_Config(void)
 {
-  RCC_OscInitTypeDef RCC_OscInitStruct = {};
-  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
```
Comment (fpistm) [inline variants/STM32G0xx/G070RBT/generic_clock.c:23]: This file should not have been modified.

**Review by fpistm — CHANGES_REQUESTED** [review]
I will deeply reviewed later. Here a first feedback.

**Review by fpistm — APPROVED** [review]
Thanks for the update.

File: variants/STM32G0xx/G070RBT/variant_NUCLEO_G070RB.cpp [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_2) != HAL_OK) {
+    Error_Handler();
+  }
+}
+
+
+#ifdef __cplusplus
+}
+#endif
+#endif /* ARDUINO_NUCLEO_G071RB */
```
Comment (ABOSTM) [inline variants/STM32G0xx/G070RBT/variant_NUCLEO_G070RB.cpp:160]:
```suggestion
#endif /* ARDUINO_NUCLEO_G070RB */
```

File: variants/STM32G0xx/G070RBT/variant_NUCLEO_G070RB.h [hunk 5]
```diff
@@ -0,0 +1,168 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2021, STMicroelectronics
```
Comment (ABOSTM) [inline variants/STM32G0xx/G070RBT/variant_NUCLEO_G070RB.h:3]:
```suggestion
 * Copyright (c) 2022, STMicroelectronics
```

**Review by ABOSTM — CHANGES_REQUESTED** [review]
2 minor comments

File: variants/STM32G0xx/G070RBT/variant_NUCLEO_G070RB.cpp [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+#endif
+
+/**
+  * @brief  System Clock Configuration
+  * @param  None
+  * @retval None
+  */
+WEAK void SystemClock_Config(void)
+{
+ RCC_OscInitTypeDef RCC_OscInitStruct = {};
```
Comment (fpistm) [inline variants/STM32G0xx/G070RBT/variant_NUCLEO_G070RB.cpp:121]: astyle issue
```suggestion
  RCC_OscInitTypeDef RCC_OscInitStruct = {};
```

**Review by fpistm — CHANGES_REQUESTED** [review]

**Review by fpistm — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Pull request merged**